sources/packages/backend/apps/api/src/route-controllers/cas-supplier/cas-supplier.aest.controller.ts

```diff
@@ -110,4 +115,39 @@ export class CASSupplierAESTController extends BaseController {
       throw error;
     }
   }
+
+  /**
+   * Retries CAS Supplier for a student.
+   * @param studentId student id.
+   * @param userToken user token.
+   */
+  @Roles(Role.AESTEditCASSupplierInfo)
+  @Post("student/:studentId/retry")
+  @ApiNotFoundResponse({
+    description: "Student not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "There is already a CAS Supplier for this student in Pending supplier verification status.",
+  })
+  async retryCASSupplier(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    @UserToken() userToken: IUserToken,
+  ): Promise<PrimaryIdentifierAPIOutDTO> {
+    const studentExist = await this.studentService.studentExists(studentId);
+    if (!studentExist) {
+      throw new NotFoundException("Student not found.");
+    }
```

Comment:
Since the service is now querying the student itself, this can be moved back to the service to avoid executing two queries for the student.